### ISSUE TITLE
使用bcc/libbpf-tools的编译环境，编译hookdetect，获取stack 信息异常

### ISSUE THREAD

**Issue by Phoexina** [pathtofile/bpf-hookdetect#3]
![image](https://user-images.githubusercontent.com/87095272/234276749-493e2e16-e59e-4eac-88e0-4174335db356.png)
循环输出stacks时发现数组中只有一个值。
![image](https://user-images.githubusercontent.com/87095272/234277517-3728cab0-6ef9-4b1d-b942-6834f68b557d.png)
但原版hookdetect应该是
![image](https://user-images.githubusercontent.com/87095272/234277659-e1ee6e6c-c6b1-4bbb-a5f9-52b9d06dce88.png)
我没有修改hookdetect原有的内核部分，但新建了别的map并在SEC("tracepoint/syscalls/sys_enter_execve")中使用。
为了契合bcc/libbpf-tools编译环境我还将ksyms改为bcc/libbpf-tools中的trace_helpers（大致看了一下代码是一样的）
请问是什么原因导致这里的stack 信息异常

**Comment by Phoexina** [comment]
Sorry for not paying attention to this is an English project.My English proficiency is limited. I used a tool to translate it, but there may be errors
title: Using the compilation environment of bcc/libbpf tools to compile hookdetect, but obtaining stack information is abnormal
When looping through stacks, it was found that there is only one value in the array.(P1 P2)
But the original hookdetect should be (P3)

I did not modify the original kernel section of hookdetect, but created a new map and used it in SEC ("tracepoint/syscalls/sys_enter_execve").
In order to fit the bcc/libbpf tools compilation environment, I also changed ksyms to trace in bcc/libbpf trace_helpers (after a rough look at the code, it looks the same)
Please help me check what may be causing the abnormal stack information here.

**Comment by Phoexina** [comment]
I solved it.

When compiling the original hookdetect, BPF_ MAP_ TYPE_ STACK_ TRACE cannot be compiled. So I modified it, but I didn't make the correct changes but never found it, so I thought the correct stack information was actually incorrect.
After moving the environment, it was discovered that the stack could not be directly obtained. 

The map should be modified as follows:

struct {
    __uint(type, BPF_MAP_TYPE_STACK_TRACE);
    __uint(max_entries, 8192);
    __uint(key_size, sizeof(u32));
    __uint(value_size, sizeof(size_t) * MAX_STACK_DEPTH);
} map_stack_traces SEC(".maps");
